cpp/src/Ice/ConnectionFactory.cpp

```diff
@@ -586,11 +594,24 @@ IceInternal::OutgoingConnectionFactory::createConnection(const TransceiverPtr& t
             throw Ice::CommunicatorDestroyedException(__FILE__, __LINE__);
         }
 
+        auto idleTimeout = _idleTimeout;
+        bool enableIdleCheck = _enableIdleCheck;
+
+        // TODO: is this the best way to detect a UDP connection?
+        if (ci.endpoint->type() == UDPEndpointType)
+        {
+            // No idle timeout or idle check for UDP connections.
+            idleTimeout = chrono::milliseconds(0);
```

Comment:
I don't think it makes sense to send heartbeats or check for idle with UDP connections.